crates/sui-types/src/messages.rs

```diff
@@ -1188,17 +1188,8 @@ impl<'a> SignatureAggregator<'a> {
 }
 
 impl CertifiedTransaction {
-    pub fn new(transaction: Transaction) -> CertifiedTransaction {
-        CertifiedTransaction {
-            transaction_digest: transaction.transaction_digest,
-            is_verified: false,
-            data: transaction.data,
-            tx_signature: transaction.tx_signature,
-            auth_sign_info: AuthorityQuorumSignInfo {
-                epoch: 0,
-                signatures: Vec::new(),
-            },
-        }
+    pub fn new(epoch: EpochId, transaction: Transaction) -> CertifiedTransaction {
```

Comment:
Is the epochID signed or just attached to a CertifiedTx?

Reply:
It's not signed. It's in fact part of the signature.

Reply:
This is interesting but also malleable and wondering if there are any side effects. In theory, it is very possible that the same quorum set of validators survive in the next epoch (our experience from other blockchains is that meaningful epoch changes happen rarely and usually only the weights are slightly recalibrated (but the pubKeys remain the same) or just one validator is updated etc). That said if the same quorum set of validators is still active one can reduce or bump the epochID in a certifiedTx object. Is there a mechanism to protect against that, is it even an issue?